### PR TITLE
Add function and plug mappings to cycle diagnostics

### PR DESCRIPTION
**Pull request by martskins**
This PR creates two new function `LanguageClient_diagnosticsNext` and `LanguageClient_diagnosticsPrevious` which enable the user to jump to the next or previous diagnostic in the current buffer, relative to the cursor's current position.

Peek 2020-07-13 17-56

### CODE REVIEW

**Comment by martskins** [comment]
@autozimu I'm not 100% sure if I missed something and there's already something in place for this or if this is in fact a missing feature, so I'll this for you to comment on and then I'll merge it if there aren't any functions that provide this functionality.

**Comment by autozimu** [comment]
I personally use quickfix to display diagnostics (the default) with `:cp`, `:cn` to jump between locations. While using location list, I believe `:prev`, `:lnext` serve similar purpose. Is this actually sufficient?

**Comment by martskins** [comment]
Not quite. The only bit that I find difficult with that workflow is that the items in the location/quickfix list are (logically) not re-sorted every time you move the cursor, so the "next" item in the location list is not really the next diagnostic based on the position of the cursor, which makes navigation a little difficult if you have many diagnostics. It can also be a little surprising, if you have the location list closed and do `:lnext` then you don't really know which diagnostic that is gonna take you too.

Another thing is that diagnostics do not seem to be ordered by line number in the location/quickfix list, maybe that can be solved though, but it still doesn't solve the previous issue.

**Comment by autozimu** [comment]
Ahh, makes sense. Can you add this gist to the function doc? Then let's
merge it in.

**Comment by martskins** [comment]
I adjusted the doc to make it a little clearer, but not sure if that's what you meant.

File: src/language_server_protocol.rs [hunk 1]
```diff
@@ -603,6 +609,37 @@ impl LanguageClient {
...
+
+            if let Some((_, diagnostic)) = diagnostics.iter_mut().find_position(|it| {
+                if direction == Direction::Next {
+                    it.range.start.line > line
+                } else {
+                    it.range.start.line < line
+                }
+            }) {
+                let target = diagnostic.range.start.line + 1;
+                let _: String = self.vim()?.rpcclient.call("execute", json!([target]))?;
```
Comment (autozimu) [inline src/language_server_protocol.rs]: Is it convenient to jump to proper column as well? Especially helpful when there are more than one diagnostics reported for the same line.

Comment (martskins) [reply src/language_server_protocol.rs]: Yeah good call!

**Pull request merged**